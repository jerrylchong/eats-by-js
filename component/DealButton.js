import React from 'react';
import {StyleSheet, Text, View, TouchableOpacity, Dimensions} from 'react-native';

const DealButton = (props) => {

    const { title, description, start, end, onPress } = props;

    return (
        <TouchableOpacity disabled={true} style = {styles.container} onPress = {onPress}>
            <View style={styles.text}>
                <Text style={styles.name}>{title}</Text>
                <Text style={styles.duration}>{start.substring(0,10)} - {end.substring(0,10)}</Text>
            </View>
            <Text style = {styles.description}>{description}</Text>
        </TouchableOpacity>
    )
}

export default DealButton;

const styles = StyleSheet.create({
    container: {
        paddingHorizontal:"3%",
        width: '100%',
        height: Dimensions.get('window').height * 0.10,
        alignSelf: 'center',
        justifyContent: 'space-evenly',
        borderBottomWidth: 0.5,
        borderColor: '#b3b3b3',
        backgroundColor: 'white'
    },
    text: {
        width: '100%',
        height: '30%',
        flexDirection: 'row',
        justifyContent: 'space-between',
        alignItems: 'center'
    },
    nameReview: {
        flexDirection: 'row',
        alignItems: 'center',
    },
    name: {
        color: '#404040',
        fontSize: 20,
        fontFamily: 'Ubuntu-Bold'
    },
    duration: {
        color: '#404040',
        fontSize: 14,
        fontFamily: 'Ubuntu'
    },
    description: {
        color: '#7e7e7e',
        fontSize: 11,
        alignSelf: 'flex-start',
        fontFamily: 'Ubuntu'
    },
})
