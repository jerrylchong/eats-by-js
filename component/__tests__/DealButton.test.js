import React from 'react';
import renderer from 'react-test-renderer';
import DealButton from "../DealButton";
import Enzyme, {shallow} from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import {Text, TouchableOpacity, View} from "react-native";

// snapshot test
test('renders correctly', () => {
    const tree = renderer.create(<DealButton title={'Title'}
                                             description={'Desc'}
                                             start={'12345678901'}
                                             end={'09876543210'}/>).toJSON();
    expect(tree).toMatchSnapshot();
})

Enzyme.configure({ adapter: new Adapter() });

test('test component rendering', () => {
    const wrapper = shallow(
        <DealButton
            title={'Title'}
            description={'Desc'}
            start={'12345678901'}
            end={'09876543210'}
        />);
    expect(wrapper.find(View)).toBeTruthy(); // checks if view is rendered
    expect(wrapper.find(TouchableOpacity)).toBeTruthy(); // checks if touchable opacity is rendered
    expect(wrapper.childAt(0).find(Text).at(0).dive().text()).toEqual('Title'); // checks if title text is rendered
    expect(wrapper.childAt(0).find(Text).at(1).dive().text()).toEqual('1234567890 - 0987654321'); // checks if cost text is rendered
    expect(wrapper.childAt(1).dive().text()).toEqual('Desc'); // checks if description text is rendered
})

test('test onPress functionality', () => {
    const onPressEvent = jest.fn();
    onPressEvent.mockReturnValue('onPress invoked'); // create mock function
    const wrapper = shallow(<DealButton onPress={onPressEvent}
                                        title={'Title'}
                                        description={'Desc'}
                                        start={'12345678901'}
                                        end={'09876543210'}/>);
    wrapper.find(TouchableOpacity).props().onPress(); // press TouchableOpacity
    expect(onPressEvent.mock.calls.length).toBe(1); // expect 1 mock function call
})