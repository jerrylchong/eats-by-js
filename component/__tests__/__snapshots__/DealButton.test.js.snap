// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessible={true}
  focusable={false}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignSelf": "center",
      "backgroundColor": "white",
      "borderBottomWidth": 0.5,
      "borderColor": "#b3b3b3",
      "height": 133.4,
      "justifyContent": "space-evenly",
      "opacity": 1,
      "paddingHorizontal": "3%",
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "height": "30%",
        "justifyContent": "space-between",
        "width": "100%",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#404040",
          "fontFamily": "Ubuntu-Bold",
          "fontSize": 20,
        }
      }
    >
      Title
    </Text>
    <Text
      style={
        Object {
          "color": "#404040",
          "fontFamily": "Ubuntu",
          "fontSize": 14,
        }
      }
    >
      1234567890
       - 
      0987654321
    </Text>
  </View>
  <Text
    style={
      Object {
        "alignSelf": "flex-start",
        "color": "#7e7e7e",
        "fontFamily": "Ubuntu",
        "fontSize": 11,
      }
    }
  >
    Desc
  </Text>
</View>
`;
